### ISSUE TITLE
Custom Training for multiple LoRAs for the same model.

### ISSUE THREAD

**Issue by bellos1203** [huggingface/peft#1571]
Hi, I'm trying to train multiple LoRA modules for my model. 
Here is my situation.
I have a large model consisting of two modules, A and B.
For input data, I have x and y, which have different characteristics from each other. 
What I want is two LoRA modules for A, which are A_x, and A_y.
Then, the output of A_x and A_y will be fed into B to generate the final output.

I found that for LoRA, the `set_adapter` function can be applied to multiple LoRA adapters.
However, it seems the `set_adapter` function is for turning back-prop on/off. 
How can I switch (or call) them in the training pipeline, without altering its learnability?

Thanks in advance :)

**Comment by BenjaminBossan** [comment]
Note that `set_adapter` enables `requires_grad` on the adapter whose name was passed and turns it off on all other adapters. However, since the PEFT model is a PyTorch `nn.Module`, as always, you can just iterate through the modules and enable or disable gradient according to your wishes:

```python
model.set_adapter("adapterA")
# now only adapter A's weights have gradients enabled
for name, param in model.named_parameters():
    if ...:  # do some check, e.g. `if "lora_" in name`
        param.requires_grad = True
```

**Comment by bellos1203** [comment]
Thank you for your answer. 
However, this isn't quite what I had in mind.

How can I set the different adapters **inside the forward path**?
For example, does the following code work? 
Because setting `requires_grad = True` inside the forward path seems little bit weird.

```
# Inside the model
def forward(self, x, y):
  model_a.set_adapter("adapter_x")
  output_x = model_a(x)
  
  model_a.set_adapter("adapter_y")
  output_y = model_a(y)
  
  final_output = model_b(output_x, output_y)
  return final_output

output = model(x, y)
loss = loss_fn(output)
loss.backward()
```

**Comment by BenjaminBossan** [comment]
It's really hard to answer your question with this little context. The code you pasted should work. It means that for `output_x`, only gradients for `adapter_x` etc. If you try this, do you get an error or does the training work?

**Comment by bellos1203** [comment]
I've tried it, and the code works :) 
I misunderstood the `set_adapter` function. 
I thought the line  `model_a.set_adapter("adapter_y")` freezes the other adapters (such as `model_a.set_adapter("adapter_x")`).
Thanks for your replies :)
